mantidproject/mantid
# Engg::Button to stop the multi-run focusing

fixes #14606 

A new button `Stop Focus` has been integrated which will abort the thread once the current focus run has been completed.

**To Test**
- Button only enabled when focusing
- When clicked `Stop Focus`, should immediately display user with appropriate warning message in the result log
- When the current focus run has been been completed; it displays warning message inside `Result Log` with last successful run & total number of focus runs that could not be processed
- When `Stop Focus` clicked during last run, it shouldn't display the final warning message inside the `Result Log`. 
- Compatible with basic `Focus`, `Focus Cropped` & `Focus Texture`
- Test with single & multi-run focusing
